src/assets/scripts/client/engine/TimeKeeper.js

```diff
@@ -16,64 +27,218 @@ class TimeKeeper {
+         * @private
+         */
+        this._futureTrackDeltaTimeCache = -1;
+
+        /**
+         * Flag used to determine if the sim has been paused
+         *
+         * This defaults to `true` because the sim is effectively paused
+         * on initial load. This way the game loop doesn't run needlessly
+         * as the sim is loading.
+         *
+         * @property _isPaused
+         * @type {boolean}
+         * @default false
+         * @private
+         */
+        this._isPaused = true;
+
+        /**
+         * Timestamp of the previous frame
+         *
+         * @property _previousFrameTimestamp
+         * @type {number}
+         */
+        this._previousFrameTimestamp = 0;
+
+        /**
+         * Timestamp for the start of rendering
+         *
+         * @property _startTimestamp
+         * @type {number}
+         * @private
+         */
+        this._startTimestamp = 0;
+
+        /**
+         * Previously known as `timewarp`, this value is used a time multiplier
+         *
+         * This value is changed via methods exposed here used within the
+         * `GameController` and is used to effectively _speed up_ this sim
+         *
+         * This value is used as a multiplier when returning the current `#_frameDeltaTime`
+         * which causes moving objects appear to have moved a farther distance than they
+         * would have at normal speed
+         *
+         * It is possible to set any numeric value via system command, though the
+         * UI enforces values of: `1`, `2` and `5` via the timewarp toggle button
+         *
+         * @property _timescale
+         * @type {number}
+         * @default 1
+         * @private
+         */
+        this._timescale = 1;
+
+        return this._init();
+    }
+
+    /**
+     * @property accumulatedDeltaTime
+     * @type {number}
+     */
+    get accumulatedDeltaTime() {
+        return this._accumulatedDeltaTime;
     }
 
     /**
      * Current timestamp in milliseconds
      *
-     * @property gameTime
-     * @return {number} current time in milliseconds
+     * This is the same value as `gameTime`.
+     * We define this getter here simply for the added context
+     * given in the name
+     *
+     * @property gameTimeMilliseconds
+     * @return {number}
      */
-    get gameTime() {
-        return (new Date()).getTime() * TIME_SECONDS_OFFSET;
+    get gameTimeMilliseconds() {
+        return (new Date()).getTime();
     }
 
     /**
+     * Current timestamp in seconds
+     *
+     * @property gameTimeMilliseconds
+     * @return {number}
+     */
+    get gameTimeSeconds() {
+        return (new Date()).getTime() * TIME.ONE_MILLISECOND_IN_SECONDS;
+    }
+
+    /**
+     * Current `#_frameDetaTime` multiplied by the current `#_timescale`.
+     *
+     * This value is capped at `100`
+     *
      * @property deltaTime
      * @return {number} current delta time in milliseconds
      */
     get deltaTime() {
```

Comment:
also since each `frame` is not like an animation frame being drawn, but rather an occurrence of the aircraft update cycle, we may even consider changing use of `frame` to something else, so wherever we see `frame`, we know it has to do with an actual re-draw.

Reply:
well, `deltaTime` has avery specific meaning in animations. I've included a link to some more info in the dock block.

In 'game' programming, `deltaTime` means exactly that; time past since the last frame.  whether that is game time or real time is irrelevant, its simply the delta between the previous time and the current time (again, whether that is game time or real time doesn't matter.)

Also, the `deltaTime` and related values are updated during each pass of the `App.update()` method which may or may not coincide with updates to the aircraft positions.  Those updates also happen within the same cycle, but I have a hunch that some are running long leading to some of our dropped frames.  All of this is fired from `requestAnimationFrame()`, hence the `frame` word used throughout.

`Frame` is typically used to define an update cycle.  In our case, calls to `App.update()` via `requestAnimationFrame()`.  Whether that results in a re-draw and render is irrelevant and up to the logic of the app.

Reply:
> In 'game' programming, deltaTime means exactly that; time past since the last frame. whether that is game time or real time is irrelevant, its simply the delta between the previous time and the current time (again, whether that is game time or real time doesn't matter.)

It is definitely worth making the distinction to which time unit this applies, because it does matter. You say it's the "time that passed since the last frame". The value of `deltaTime` is used by the physics to determine how far the aircraft moved, based on the speed and length of time since the last frame. The time of interest is the _game time_ that has passed, not the difference between and old and new `Date().getTime()`. The latter properly takes into account any timewarping, while the latter would be more useful for showing an on-screen framerate, for example.

For example, let's say we are operating with a 2x timescale. If the `deltaTime` is in _game time_, or in _real time_, it makes a difference. When we go to make the physics calculations, we want the _game time_ that has passed, aka the `real time X timescale`. So whether `deltaTime` is already taking into account the timescale or not is very important. Without knowing, we could easily over-apply or under-apply the timescale adjustment, which we certainly want to avoid.

Reply:
I smell what you're stepping in.  We account for both (which perhaps warrants a little more documentation).

`#_frameDeltaTime` is strictly the change in timestamps from one frame to the next.  Always, no matter `#_simulationRate` or `#_isPaused`.

`#deltaTime` takes simulation rate into account and returns an appropriate `#deltaTime`:
```javascript
get deltaTime() {
    const deltaTimeOffsetBySimulationRate = this._frameDeltaTime * this._simulationRate;

    return Math.min(deltaTimeOffsetBySimulationRate, 100);
}
```

Reply:
Its important to note that as far as anything other than the `TimeKeeper` is concerned, `deltaTime` is the product of `_#frameDeltaTime` and `#_simulationRate`.  In very few circumstances does anything need to know, or care, about what the value of `#_simulationRate` actually is.

Reply:
@erikquinn documentation added

Reply:
perfect, that's the distinction I was looking for. Thanks!